File: src/main/java/me/ag2s/epublib/domain/ResourceInputStream.java
package me.ag2s.epublib.domain;

import java.io.FilterInputStream;
import java.io.IOException;
import java.io.InputStream;
import java.util.zip.ZipFile;

public class ResourceInputStream extends FilterInputStream {
   private final ZipFile zipFile;

   public ResourceInputStream(InputStream in, ZipFile zipFile) {
      super(in);
      this.zipFile = zipFile;
   }

   public void close() throws IOException {
      super.close();
      this.zipFile.close();
   }
}

File: src/main/kotlin/me/ag2s/epublib/domain/ResourceInputStream.kt
package me.ag2s.epublib.domain

import java.io.FilterInputStream
import java.io.IOException
import java.io.InputStream
import java.util.zip.ZipFile

class ResourceInputStream(input: InputStream, private val zipFile: ZipFile) : FilterInputStream(input) {
    @Throws(IOException::class)
    override fun close() { super.close(); zipFile.close() }
}
